lib/apache_svn_url_parser.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+class ApacheSvnUrlParser < URLParser
+  private
+
+  def full_domain
+    'https://svn.apache.org/viewvc'
+  end
+
+  def tlds
+    %w(org)
+  end
+
+  def domain
+    'svn.apache'
+  end
```

Comment:
All of this is sort of awkward because the structure of the URLs doesn't really match any of the other providers.